plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java

```diff
@@ -74,25 +80,29 @@ public static IntegrationCloudServiceClient getInstance() {
      * @throws NetworkUnavailableException
      */
     public boolean login(String username, String password, String tenant)
-            throws CloudDeploymentException, InvalidTokenException, NetworkUnavailableException, HttpClientException {
+            throws CloudDeploymentException, NetworkUnavailableException, HttpClientException {
         String loginUrl = CloudServiceConstants.ServiceEndpoints.LOGIN_URL;
 
         // Integration Cloud expects the username to be in <email>@<organization_key> format
         String user = username + "@" + tenant;
 
         Map<String, String> data = new HashMap<>();
-        data.put("action", CloudServiceConstants.Actions.LOGIN);
-        data.put("userName", user);
-        data.put("password", password);
-
-        String response = HTTPClientUtil.sendPostWithFormData(loginUrl, new HashMap<String, String>(), data,
-                cookieStore);
-
-        JsonParser parser = new JsonParser();
-        JsonElement jsonResponse = parser.parse(response);
+        data.put(CloudServiceConstants.Parameters.PARAM_ACTION, CloudServiceConstants.Actions.LOGIN);
+        data.put(CloudServiceConstants.Parameters.PARAM_USERNAME, user);
+        data.put(CloudServiceConstants.Parameters.PARAM_PASSWORD, password);
+
+        try {
+            String response = HTTPClientUtil.sendPostWithFormData(loginUrl, new HashMap<String, String>(), data,
+                    cookieStore);
+            JsonParser parser = new JsonParser();
+            JsonElement jsonResponse = parser.parse(response);
+            String message = jsonResponse.getAsJsonObject().get("message").getAsString();
```

Comment:
Don't we have to check null here? Can these become null?

Reply:
Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/1063